spec/features/idv/doc_auth/document_capture_spec.rb

```diff
@@ -298,9 +308,353 @@
         end
 
         context 'selfie with error is uploaded' do
+          before do
+            allow(IdentityConfig.store).to receive(:doc_auth_max_attempts).and_return(99)
+
+            allow_any_instance_of(FederatedProtocols::Oidc).
+              to receive(:biometric_comparison_required?).
+              and_return(true)
+            perform_in_browser(:mobile) do
+              visit_idp_from_oidc_sp_with_ial2(biometric_comparison_required: true)
+              sign_in_and_2fa_user(@user)
+              complete_doc_auth_steps_before_document_capture_step
+            end
+          end
+
+          it 'shows the correct error message for the given error' do
+            # when the only error is a doc auth error
+
+            perform_in_browser(:mobile) do
+              attach_images(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_doc_auth_fail_selfie_pass.yml'
+                ),
+              )
+              attach_selfie(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_doc_auth_fail_selfie_pass.yml'
+                ),
+              )
+
+              submit_images
+
+              h1_error_message = strip_tags(t('errors.doc_auth.rate_limited_heading'))
+              expect(page).to have_content(h1_error_message)
+
+              body_error_message = strip_tags(t('doc_auth.errors.dpi.top_msg'))
+              expect(page).to have_content(body_error_message)
+
+              click_try_again
+              expect(page).to have_current_path(idv_document_capture_path)
+
+              inline_error_message = strip_tags(t('doc_auth.errors.dpi.failed_short'))
+              expect(page).to have_content(inline_error_message)
+
+              expect(page).to have_current_path(idv_document_capture_url)
+
+              # when doc auth result passes but liveness fails
+
+              attach_images(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_no_liveness.yml'
+                ),
+              )
+              attach_selfie(
+                Rails.root.join(
+                  'spec', 'fixtures',
+                  'ial2_test_credential_no_liveness.yml'
+                ),
+              )
+
+              submit_images
+
+              h1_error_message = strip_tags(
+                t('errors.doc_auth.selfie_not_live_or_poor_quality_heading'),
+              )
+              expect(page).to have_content(h1_error_message)
+
+              body_error_message = strip_tags(t('doc_auth.errors.alerts.selfie_not_live'))
+              expect(page).to have_content(body_error_message)
+
+              click_try_again
+              expect(page).to have_current_path(idv_document_capture_path)
+
+              # inline error to be fixed in lg-12999
+
+              # when there are both doc auth errors and liveness errors
```

Comment:
Are we expecting that only front/back should have errors here? I would naively expect all three?

Reply:
I think this is the same case of the inline errors. I agree that we should test all 3 but we know this is broken until LG-12999 is fixed. So I could either add in expectations of what we know is currently displaying (but is wrong), or what I did here which is leave a note. I think fixing it would be out of scope.